src/components/connections/Connection.ts

```diff
@@ -0,0 +1,177 @@
+import * as vscode from "vscode";
+import { Engine as QueryEngine } from "../Query";
+import { INode } from "./INode";
+import { Status } from "./Status";
+import { ConnectionNode, InfluxDBConectionsKey } from "./ConnectionNode";
+
+const uuidv1 = require("uuid/v1");
+export interface InfluxDBConnection {
+  readonly name: string;
+  readonly hostNport: string;
+  readonly token: string;
+  readonly org: string;
+}
+
+export class InfluxDBTreeDataProvider
+  implements vscode.TreeDataProvider<INode> {
+  public _onDidChangeTreeData: vscode.EventEmitter<
+    INode
+  > = new vscode.EventEmitter<INode>();
+
+  public readonly onDidChangeTreeData: vscode.Event<INode> = this
+    ._onDidChangeTreeData.event;
+
+  constructor(
```

Comment:
why do we need this constructor if it isn't doing anything?

Reply:
Is that a model like https://riptutorial.com/typescript/example/5068/constructors ?

Reply:
a class doesn't require a constructor, if you omit this constructor you can still instantiate it.